### PR TITLE
Misc POM improvements

### PR DESCRIPTION
**Pull request by dkwasny-ibm**
* Adding configuration for the flattener plugin
* Moving `dependencyManagement` info from `cohort-engine/pom.xml` to the parent pom
* Adding the `<scm>` block to all poms

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -73,6 +71,117 @@
 		</repository>		
 	</repositories>
 
+	<dependencyManagement>
```
Comment (jillrdoty) [inline pom.xml:74]: Is there a reason for moving these dependencies?  My initial thought on leaving them was that they weren't necessarily needed by every sub-project and shouldn't be included if not.

Comment (dkwasny-ibm) [reply pom.xml:74]: This is the `<dependencyManagement>` block, which is effectively the "BOM" of the project.  Corey set up these versions to line up with the CQL engine and its dependencies back when `cohort-engine` was the only project.  These shouldn't force any dependencies on any subprojects, just the versions.

Comment (jillrdoty) [reply pom.xml:74]: It feels a little weird to call out the versions specifically above *and* have the dependencyManagement block... but I guess I don't care that much.

Comment (jillrdoty) [reply pom.xml:74]: Kwas explained that the versions are used for multiple dependencies, so I take it back.

**Review by jillrdoty — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (csandersdev) [reply pom.xml:74]: Yep. This is very important. I already ran into a case where cql-ruler was pulling in a different version of the engine than we were intending to use. Dependency management smooths it all out.

**Review by csandersdev — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (msargentibm) [reply pom.xml:74]: Thanks for the discussion. Had a similar train of thought as Jill, but this cleared things up.

**Review by msargentibm — APPROVED** [review]

**Pull request merged**